packages/tc-ui/src/pages/edit/template.jsx

```diff
@@ -31,10 +31,23 @@ const PropertyInputs = ({ fields, data, type, assignComponent, hasError }) => {
 				? data[`${propertyName}_rel`] || data[propertyName]
 				: data[propertyName];
 
+			/**
+			 * Sending the entire record to the client multiple times can increase the payload immensely
+			 * However some components rely on this (eg "decommissioned" functionality)
+			 * As a workaround for now, truncate long text fields
+			 * TODO: think of a better way to be selective about what we send
+			 */
+			const clientSideRecord = {};
+			Object.entries(data).forEach(([key, value]) => {
+				clientSideRecord[key] =
```

Comment:
a nicer test would be `properties[key].type !== 'Document'`

Reply:
I though a schema driven approach would be good, but didn't really understand the different between `fields`, `properties` and `propertyfields`.

You sure it shouldn't be `fields[key].type`?